File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* r) {
        if(!r) return 0;
        queue<TreeNode*> q;
        q.push(r);
        int ans{};
        while(!q.empty()) {
            int size = q.size();
            for(int i = 0 ; i  < size; ++ i) {
                auto cur = q.front();
                q.pop();
                if(cur->left) q.push(cur->left);
                if(cur->right) q.push(cur->right);
            }
            ++ans;
        }
        return ans;
    }
};

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode], d = 0) -> int:
        if root is None:
            return 0
            
        l, r = 0, 0
        if root.left:
            l = self.maxDepth(root.left, d + 1)
        if root.right:
            r = self.maxDepth(root.right, d + 1)
        return 1 + max(l, r)
        